### PR TITLE
StateStore API implementation for CockroachDB

### PR DESCRIPTION
**Pull request by r-c-correa**
# Description

This PR introduces a CockroachDB StateStore component. The code is implemented following the example of the PostgreSQL.

## Issue reference

Please reference the issue this PR will close: #1556

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [x] Created/updated tests
* [x] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#2237

### CODE REVIEW

**Comment by r-c-correa** [comment]
Next commit I will send Conformance Tests for CockroachDB 😃

**Comment by yaron2** [comment]
Really excited to see this! Great contribution and kudos for adding conformance tests.

File: state/cockroachdb/cockroachdb_access.go
Comment (Taction) [inline state/cockroachdb/cockroachdb_access.go:31]: nits: PostgreSQL here should be CockroachDB

**Comment by Taction** [comment]
@r-c-correa Thanks for your contribution! Looks good for me overall, just some small changes.

File: state/cockroachdb/cockroachdb_access.go
Comment (mukundansundar) [inline state/cockroachdb/cockroachdb_access.go:252]: do check for transaction ordering.

**Review by mukundansundar — COMMENTED** [review]
@r-c-correa Thanks for the contribution. Added some comments. 
Please look at this PR https://github.com/dapr/components-contrib/pull/1550, for conformance with respect to transaction ordering.

**Comment by yaron2** [comment]
@r-c-correa ping. address the changes and we can get this merged.

**Comment by r-c-correa** [comment]
@yaron2 I did the changes, I have only one question, waiting for @mukundansundar answer 😄

**Comment by mukundansundar** [comment]
> Please look at this PR #1550, for conformance with respect to transaction ordering.

@r-c-correa 
If this is with respect to the line above ...
Then the transaction order conformance tests pass for cockroachDB ... 
https://github.com/dapr/components-contrib/runs/5631808447?check_suite_focus=true#step:33:219

File: state/cockroachdb/cockroachdb_access.go
Comment (yaron2) [reply state/cockroachdb/cockroachdb_access.go:137]: ping @r-c-correa

Comment (yaron2) [reply state/cockroachdb/cockroachdb_access.go:137]: @mukundansundar there is a possibility for the error to not be an etag error, but there's no way of knowing for certain. that's how it's been done in other state stores, and why the error message that comes back to the user is non-deterministic in it's description (ie. "possible etag mismatch").

**Review by yaron2 — DISMISSED** [review]

**Comment by mukundansundar** [comment]
updated doc PR ... https://github.com/dapr/docs/pull/2276

File: state/cockroachdb/cockroachdb_access.go
Comment (mukundansundar) [inline state/cockroachdb/cockroachdb_access.go:472]: nit: in delete operation

**Review by mukundansundar — COMMENTED** [review]
@r-c-correa small change. otherwise looks good.

**Review by mukundansundar — APPROVED** [review]
LGTM

**Comment by mukundansundar** [comment]
@r-c-correa Thanks for this awesome contribution. Please do make sure that the docs are accurate and follow up on that PR.

**Comment by mukundansundar** [comment]
@r-c-correa ... There needs to be a follow up PR to integrate the component into dapr ... Since it is a new component ...
Once this PR is merged, follow from point 3 .... https://github.com/dapr/components-contrib/blob/master/docs/developing-component.md#submit-your-component

**Comment by mukundansundar** [comment]
> Of course, I do that @mukundansundar 😄
> I can use that [PR](https://github.com/dapr/dapr/pull/4335) or need to create another?

You would need to update components-contirb ref ... But that should suffice ....

**Pull request merged**
